src/main/kotlin/com/reicheltp/celtic_rituals/proxy/ClientProxy.kt

```diff
@@ -27,5 +32,13 @@ class ClientProxy : CommonProxy() {
             RitualBagEntity::class.java,
             RitualBagRenderer.RitualBagRenderFactory()
         )
+
+        Minecraft.getInstance().itemColors.register(IItemColor { item: ItemStack, layer: Int ->
+            if (layer == 0) {
+                -1
+            } else {
+                RitualBagItem.getColor(item)
+            }
+        }, ModItems.RITUAL_BAG)
```

Comment:
It would be very helpful for me as a beginner, when there would be some comments in the code for such "complex" logic.
Or maybe breaking down the logic with more variables would also help.

Reply:
I guess you are registering some color for the ritual bag by getting the color from the item?

Reply:
but why `-1`?

Reply:
We have 2 layers that render the item. The first is the bag, the second the overlay. We do not apply a color blending to the first one. That's why -1 (Do not recolor).

The second layer the overlay is colored depending on the stack. This logic is moved to the ritual bag.